pkg/syncer/cnsoperator/util/util.go

```diff
@@ -140,3 +149,29 @@ func GetTKGVMIP(ctx context.Context, vmOperatorClient client.Client, dc dynamic.
 	log.Infof("Found external IP Address %s for VirtualMachine %s/%s", ip, vmNamespace, vmName)
 	return ip, nil
 }
+
+// GetNetworkProvider reads the network-config configmap in Supervisor cluster and
+// returns the network provider as NSX-T or VDS. It returns an empty string if network provider
+// is not present in the configmap.
+func GetNetworkProvider(ctx context.Context) (string, error) {
+	log := logger.GetLogger(ctx)
+	k8sclient, err := k8s.NewClient(ctx)
+	if err != nil {
+		log.Errorf("GetNetworkProvider: Creating Kubernetes client failed. Err: %v", err)
+		return "", err
+	}
+	cm, err := k8sclient.CoreV1().ConfigMaps(kubeSystemNamespace).Get(ctx, wcpNetworkConfigMap, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to get config map %q in namespace %q", wcpNetworkConfigMap, kubeSystemNamespace)
+	}
+
+	for k, v := range cm.Data {
+		if k == networkProvider {
+			log.Debugf("Returning network provider value as: %q", v)
+			return v, nil
+		}
+	}
+
+	log.Errorf("could not find network provider field in configmap %q  in namespace %q", wcpNetworkConfigMap, kubeSystemNamespace)
+	return "", nil
```

Comment:
I think we should return an error if the network provider field is empty.

Reply:
Agree with @RaunakShah

Reply:
Changed.